mm2src/mm2_main/src/database/stats_swaps.rs

```diff
@@ -100,6 +100,13 @@ pub const ADD_MAKER_TAKER_GUI_AND_VERSION: &[&str] = &[
 
 pub const SELECT_ID_BY_UUID: &str = "SELECT id FROM stats_swaps WHERE uuid = ?1";
 
+pub const ADD_SWAP_VERSION_AND_MARKET_MARGIN: &[&str] = &[
+    "ALTER TABLE stats_swaps ADD COLUMN swap_version INTEGER NOT NULL DEFAULT 0;",
+    "ALTER TABLE stats_swaps ADD COLUMN market_margin DECIMAL;",
+    // NULL = unknown, 0 = not a bot, 1 = bot
+    "ALTER TABLE stats_swaps ADD COLUMN is_maker_bot INTEGER;",
```

Comment:
I wonder if making it `maker_type` with backing it Rust enum would be any good. :thinking: Just for the extensibility.


So it would look something like this:

```
enum MakerType {
  Unknown,
  WebUser,
  DesktopUser,
  Bot,
  .
  . 
  .
}
```

and can be extended to anything without changing things on the DB side.

Reply:
> I wonder if making it maker_type with backing it Rust enum would be any good.

like the idea. we can have a new issue about that to harvest such data (cc/ @shamardy)

we can ofc do a migration to change the table to the appropriate format later.